js/permissionsSettings.js

```diff
@@ -0,0 +1,65 @@
+(function ($)
+  {
+    $(document).ready(function() {
+
+      var $area = $('#premisAccessPermissionsArea');
+
+      $area.find('.cbx')
+        .hover(function () {
+          $(this).addClass('hover');
+        }, function () {
+          $(this).removeClass('hover');
+        })
+        .on('click', function (event) {
+          if (event.target.tagName !== 'INPUT') {
+            $(event.target).find('input').trigger('click');
+          }
+        });
+
+      $area.find('.all').on('click', function (event) {
+        event.preventDefault();
+        $area.find('input').prop('checked', true);
+      });
+
+      $area.find('.none').on('click', function (event) {
+        event.preventDefault();
+        $area.find('input').prop('checked', false);
+      });
+
+      $area.find('.btn-check-col').on('click', function (event) {
+
+        event.preventDefault();
+
+        // Find index of the column being clicked
+        var $li = $(this.parentNode);
+        var index;
+        $area.find('th.premis-permissions-mrt li').each(function (i) {
+          if ($li.is(this)) {
+            index = i;
+            return false;
+          }
+        });
+
+        // Create a jQuery object with all the checkboxes of the column in it
+        var cbs = jQuery();
```

Comment:
Right! It's a good practice to remember that we are dealing with jQuery objects. I'll fix it.